### PR TITLE
feat(cli): add multi-agent installation support

### PR DESCRIPTION
**Pull request by jpoley**
## Summary
- Implement multi-agent installation support for `specify init`
- Users can now select multiple AI assistants interactively or via comma-separated `--ai` flag
- Full backward compatibility maintained - single agent usage unchanged
- Comprehensive test coverage with 28 new tests (294 total tests passing)

### Core Functionality
- **parse_agent_list()**: Parse comma-separated agent input with edge case handling
- **select_multiple_with_checkboxes()**: Interactive multi-select UI using Space/Enter controls
- **init() command**: Accepts `--ai claude,copilot,cursor-agent` syntax
- **download functions**: Support both single agent (str) and multiple agents (list[str])
- **Tool checking**: Validates all selected CLI-based agents before installation

### User Experience
- Interactive mode shows multi-select UI with helpful tip
- Security notices display all selected agent folders
- Agent-specific setup (e.g., Codex CODEX_HOME) works with multiple agents
- Clear output showing all selected agents

### Testing
- 28 new unit tests covering parse_agent_list(), validation, combinations, tool checks
- All edge cases tested: empty input, trailing commas, whitespace, etc.
- Backward compatibility tests ensure single agent usage still works
- Agent folder uniqueness tests prevent conflicts
- All 294 tests passing

### Documentation
- Updated README.md with multi-agent examples
- Added "Multi-Agent Support" section with benefits and usage
- Updated CLI help text to mention comma-separated syntax
- Updated docstring examples in init()

## Test plan
- [x] `specify init my-project --ai claude` works (single agent backward compat)
- [x] `specify init my-project --ai claude,copilot` installs both templates
- [x] `specify init my-project --ai claude,cursor-agent,copilot` installs three agents
- [x] Interactive mode allows multi-select with Space/Enter
- [x] All 294 tests pass: `pytest tests/ -q`
- [x] Linting clean: `ruff check` and `ruff format`
- [x] Invalid agents are detected and rejected with clear error messages
- [x] Tool checking validates all CLI-based agents (claude, gemini, etc.)
- [x] Security notices show all agent folders

## Breaking Changes
None - fully backward compatible

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/specify_cli/__init__.py [hunk 1]
```diff
@@ -1163,15 +1326,25 @@ def download_and_extract_template(
...
+    their respective agent directories are all extracted to the project.
+
     Returns project_path. Uses tracker if provided (with keys: fetch, download, extract, cleanup)
     """
+    # Normalize to list for consistent handling
+    if isinstance(ai_assistants, str):
+        ai_assistants = [ai_assistants]
+
+    # Use first agent for template download (templates contain all agent directories)
+    primary_agent = ai_assistants[0]
```
[the remark this answers is not in the document]
Comment (jpoley) [reply src/specify_cli/__init__.py]: agree

File: src/specify_cli/__init__.py [hunk 2]
```diff
@@ -1163,15 +1326,25 @@ def download_and_extract_template(
...
+
+    Supports single or multiple AI assistants. When multiple assistants are specified,
+    their respective agent directories are all extracted to the project.
+
     Returns project_path. Uses tracker if provided (with keys: fetch, download, extract, cleanup)
     """
+    # Normalize to list for consistent handling
+    if isinstance(ai_assistants, str):
+        ai_assistants = [ai_assistants]
+
```
[the remark this answers is not in the document]
Comment (jpoley) [reply src/specify_cli/__init__.py]: lets validate this

File: src/specify_cli/__init__.py [hunk 3]
```diff
@@ -492,6 +492,157 @@ def run_selection_loop():
...
+        ['claude']
+    """
+    if not agent_str or not agent_str.strip():
+        return []
+
+    # Split by comma and strip whitespace from each agent name
+    agents = [agent.strip() for agent in agent_str.split(",")]
+
+    # Filter out empty strings
+    return [agent for agent in agents if agent]
```
[the remark this answers is not in the document]
Comment (jpoley) [reply src/specify_cli/__init__.py]: agree, lets fix, even if it seems more sloppy than harmful, should be idempotent

File: src/specify_cli/__init__.py [hunk 4]
```diff
@@ -933,7 +1084,8 @@ def _pick_latest(releases: list[dict]) -> Optional[dict]:
 
 def download_and_extract_two_stage(
     project_path: Path,
-    ai_assistant: str,
+    ai_assistants: list[str]
+    | str,  # Support both list and single string for backward compatibility
```
[the remark this answers is not in the document]
Comment (jpoley) [reply src/specify_cli/__init__.py]: agree, lets fix

File: src/specify_cli/__init__.py [hunk 5]
```diff
@@ -1149,7 +1311,8 @@ def download_and_extract_two_stage(
 
 def download_and_extract_template(
     project_path: Path,
-    ai_assistant: str,
+    ai_assistants: list[str]
+    | str,  # Support both list and single string for backward compatibility
```
[the remark this answers is not in the document]
Comment (jpoley) [reply src/specify_cli/__init__.py]: agree, lets fix

File: src/specify_cli/__init__.py [hunk 6]
```diff
@@ -946,12 +1098,22 @@ def download_and_extract_two_stage(
...
     """
+    # Normalize to list for consistent handling
+    if isinstance(ai_assistants, str):
+        ai_assistants = [ai_assistants]
     current_dir = Path.cwd()
     base_zip = None
     ext_zip = None
 
+    # Use first agent for template download (templates contain all agent directories)
+    primary_agent = ai_assistants[0]
```
[the remark this answers is not in the document]
Comment (jpoley) [reply src/specify_cli/__init__.py]: agree, lets validate (and remember to always validate when possible) defensive coding

File: tests/test_multi_agent.py [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+        """Test parsing with multiple consecutive commas."""
+        result = parse_agent_list("claude,,copilot")
+        assert result == ["claude", "copilot"]
+
+    def test_all_supported_agents(self):
+        """Test parsing all supported agents."""
+        all_agents = ",".join(AGENT_CONFIG.keys())
+        result = parse_agent_list(all_agents)
+        assert result == list(AGENT_CONFIG.keys())
+
```
[the remark this answers is not in the document]
Comment (jpoley) [reply tests/test_multi_agent.py]: agree

**Comment by jpoley** [comment]
we need these fixed, then resubmitted
